### DISCUSSION TITLE
Node for click on NodeWithToolBar

### DISCUSSION THREAD

**Discussion by mjcee** [xyflow/xyflow#4745] [Q&A] [answered]
How do I get the node id of the node associated with a NodeWithToolBar from the parameter for a click event

**Comment by moklick** [comment, marked as answer]
If you are speaking of the NodeToolbar example, you get the node id via props:

```js
// id is the node id here
function NodeWithToolbar({ data, id }) {
```

**Reply by mjcee** [reply]
That's what I needed.  Thanks for your prompt reply.
